app/service/data_svc.py

```diff
@@ -150,6 +150,12 @@ async def load_source_file(self, filename, access):
             source.access = access
             await self.store(source)
 
+    async def load_objective_file(self, filename, access):
+        for src in self.strip_yml(filename):
+            obj = Objective.load(src)
+            obj.access = access
+            await self.store(obj)
+
```

Comment:
load_adversary_file / load_source_file / load_objective_file is all the same function. We can abstract here and minimize the repetition